skbase/base/_base.py

```diff
@@ -914,12 +914,20 @@ def check_is_fitted(self):
                 f"Please call `fit` first."
             )
 
-    def get_fitted_params(self):
+    def get_fitted_params(self, deep=True):
         """Get fitted parameters.
 
         State required:
             Requires state to be "fitted".
 
+        Parameters
+        ----------
+        deep : bool, default=True
+            If True, will return fitted parameters for this estimator,
```

Comment:
Maybe have this setup to use bulleted list?

How about:

```python
def get_fitted_params(self, deep=True):
    """Get fitted parameters.

    Requires estimator to be fitted.

    Parameters
    ----------
    deep : bool, default=True
        Whether to return fitted parameters of components.

        - If True, will return fitted parameters for this estimator, plus parameters of components.
        - If False, only returns the fitted parameters for this estimator.       
    
    Returns
    -------
    fitted_params : dict of fitted parameters
        Dictionary where keys are str names of fitted parameters and values are the fitted values.
        Parameters of components are indexed as [componentname]__[paramname].
    """    
```

Reply:
Note that I also suggested minor updates to returns.

Reply:
Re the update to returns: think it's worth pointing out the recursion explicitly.

If you go over the abridged form, it is not clear to a reader whether recursion happens or not - strictly speaking, taking the description at face value with brutal mathematical precision, recursion would *not* occur, unless it is additionally stated.

Reply:
apologies - I realize my answer must not make sense, since I have mixed up the `get_params` pull request with this one, in my memory.

I have now adopted your suggestions, and also improved description coming from the `get_params` PR (which includes your suggestion for the return).